File: backend/app/repositorios/MensajesSQLiteRepo.go
package repositorios

import(
	"database/sql"
	"mizcuin/modelos"
	"os"
	"fmt"
	_ "github.com/mattn/go-sqlite3"
)

var dbPathMensajes string

func init(){
	dbPathMensajes = os.Getenv("DB_PATH")
}

func GuardarMensaje(consulta string, mensaje modelos.Mensajes) bool{
	db, err := sql.Open("sqlite3", dbPathMensajes)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en mensajes", err)
		return false
	}

	defer db.Close()

	_, err = db.Exec(consulta, mensaje.Id_usuario_envia, mensaje.Id_usuario_recibe, mensaje.Id_trabajo, mensaje.Contenido, mensaje.Fecha_mensaje)
	if err != nil {
		fmt.Println("Error al ejecutar la consulta en mensajes", err)
		return false
	}

	fmt.Println("Mensaje guardado correctamente: ", consulta)
	return true
	
}

func BuscarMensajes(consulta string, id_usuario int, id_trabajo int) (modelos.ListadoMensajes, bool){
	db, err := sql.Open("sqlite3", dbPathMensajes)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en mensajes", err)
		return modelos.ListadoMensajes{}, false
	}

	defer db.Close()

	rows, err := db.Query(consulta, id_usuario, id_trabajo)

	if err != nil {
		fmt.Println("Error al buscar el mensaje", err)
		return modelos.ListadoMensajes{}, false
	}

	defer rows.Close()

	var mensajes []modelos.Mensajes

	for rows.Next() {
		var idMensaje int
		var idUsuarioEnvia int
		var idUsuarioRecibe int
		var idTrabajo int
		var contenido string
		var fechaMensaje string

		err = rows.Scan(&idMensaje, &idUsuarioEnvia, &idUsuarioRecibe, &idTrabajo, &contenido, &fechaMensaje)

		if err != nil {
			fmt.Println("Error al buscar el mensaje", err)
			return modelos.ListadoMensajes{}, false
		}

		mensajes = append(mensajes, modelos.Mensajes{Id_mensaje: idMensaje, Id_usuario_envia: idUsuarioEnvia, Id_usuario_recibe: idUsuarioRecibe, Id_trabajo: idTrabajo, Contenido: contenido, Fecha_mensaje: fechaMensaje})
	}

	fmt.Println("Mensaje encontrado correctamente: ", consulta)
	return modelos.ListadoMensajes{Mensajes: mensajes}, true
}

func ActualizarMensaje(consulta string, id_mensaje int, contenido string) bool{
	db, err := sql.Open("sqlite3", dbPathMensajes)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en mensajes", err)
		return false
	}

	defer db.Close()

	_, err = db.Exec(consulta, id_mensaje, contenido)
	if err != nil {
		fmt.Println("Error al ejecutar la consulta en mensajes", err)
		return false
	}

	fmt.Println("Mensaje actualizado correctamente: ", consulta)
	return true
}

func EliminarMensaje(consulta string, id_mensaje int) bool{
	db, err := sql.Open("sqlite3", dbPathMensajes)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en mensajes", err)
		return false
	}

	defer db.Close()

	_, err = db.Exec(consulta, id_mensaje)
	if err != nil {
		fmt.Println("Error al ejecutar la consulta en mensajes", err)
		return false
	}

	fmt.Println("Mensaje eliminado correctamente: ", consulta)
	return true
}

File: backend_Python/app/repositorios/mensajesSQLiterepo.py
import sqlite3
import os
# Asegúrate de que el archivo se llame Mensajes.py y la clase Mensaje
from modelos.Mensajes import Mensaje

DB_PATH_MENSAJES = os.getenv("DB_PATH")

def guardar_mensaje(consulta: str, mensaje: Mensaje) -> bool:
    try:
        with sqlite3.connect(DB_PATH_MENSAJES) as conn:
            cursor = conn.cursor()
            # Aquí 'mensaje' sí existe porque es un parámetro de esta función
            valores = (
                mensaje.id_animalLover_emisor, 
                mensaje.id_animalLover_receptor, 
                mensaje.id_trabajo, 
                mensaje.contenido, 
                mensaje.fecha_mensaje
            )
            cursor.execute(consulta, valores)
            conn.commit()
        return True
    except Exception as e:
        print(f"Error en guardar_mensaje: {e}")
        return False

def buscar_mensajes(consulta: str, id_animalLover: int, id_trabajo: int) -> tuple[list[Mensaje], bool]:
    try:
        with sqlite3.connect(DB_PATH_MENSAJES) as conn:
            cursor = conn.cursor()
            cursor.execute(consulta, (id_animalLover, id_trabajo))
            rows = cursor.fetchall()
            
            listado_mensajes = []
            for row in rows:
                # LINEA 41: Usamos la Clase 'Mensaje' (Mayúscula) para crear el objeto
                nuevo_mensaje = Mensaje(*row) 
                listado_mensajes.append(nuevo_mensaje)
                
        return listado_mensajes, True
    except Exception as e:
        print(f"Error en buscar_mensajes: {e}")
        return [], False
def actualizar_mensaje(consulta: str, id_mensaje: int, contenido: str) -> bool:
    """
    Actualiza el contenido de un mensaje existente.
    """
    try:
        with sqlite3.connect(DB_PATH_MENSAJES) as conn:
            cursor = conn.cursor()
            cursor.execute(consulta, (contenido, id_mensaje))
            conn.commit()
            
        print(f"Mensaje actualizado correctamente: {consulta}")
        return True
    except Exception as e:
        print(f"Error al ejecutar la actualización: {e}")
        return False

def eliminar_mensaje(consulta: str, id_mensaje: int) -> bool:
    """
    Elimina un mensaje de la base de datos por su ID.
    """
    try:
        with sqlite3.connect(DB_PATH_MENSAJES) as conn:
            cursor = conn.cursor()
            cursor.execute(consulta, (id_mensaje,))
            conn.commit()
            
        print(f"Mensaje eliminado correctamente: {consulta}")
        return True
    except Exception as e:
        print(f"Error al ejecutar la eliminación: {e}")
        return False